lib/workers/repository/process/lookup/index.ts

```diff
@@ -104,9 +104,12 @@ export async function lookupUpdates(
 
       const latestVersion = dependency.tags?.latest;
       // Filter out any results from datasource that don't comply with our versioning
-      let allVersions = dependency.releases.filter((release) =>
-        versioning.isVersion(release.version)
-      );
+      let allVersions = dependency.releases
+        .filter((release) => versioning.isVersion(release.version))
+        .map((release) => {
+          release.registryUrl ??= dependency.registryUrl;
```

Comment:
Why populate every release? I would prefer to populate it only if it differs from the primary result url

Reply:
Based on `mergeRegistries` function, registry URL will be mostly logged as part of the main object.

The only case where we will have the URL specified as part of the release is when we were able to fetch releases from multiple registry URLs. in this case the main object will have no URLs specified, as it is getting deleted as part of `mergeRegistries` function.